src/pudl/transform/eia923.py

```diff
@@ -1390,3 +1391,148 @@ def cooling_system_information_continuity(csi):
+@asset(io_manager_key="pudl_io_manager")
+def _core_eia923__fgd_operation_maintenance(
+    raw_eia923__fgd_operation_maintenance: pd.DataFrame,
+) -> pd.DataFrame:
+    """Transforms the _core_eia923__fgd_operation_maintenance table.
+
+    Transformations include:
+
+    * Drop values with plant and boiler id values of NA.
+    * Replace . values with NA.
+    * Convert thousands of dollars to dollars.
+    * Fix datetimes for SO2 test dates.
+    * Ensure a unique primary key and drop some duplicated rows.
+
+    Args:
+        raw_eia923__fgd_operation_maintenance: The raw ``raw_eia923__fgd_operation_maintenance`` dataframe.
+
+    Returns:
+        Cleaned ``_core_eia923__fgd_operation_maintenance`` dataframe ready for harvesting.
+    """
+    fgd_df = raw_eia923__fgd_operation_maintenance
+
+    # Drop duplicate and empty rows to ensure the primary key is unique.
+    fgd_df = fgd_df.dropna(subset=["plant_id_eia", "so2_control_id_eia"])
+    fgd_df = fgd_df.drop_duplicates()
+
+    # Replace the EIA923 NA value ('.') with a real NA value.
+    fgd_df = pudl.helpers.fix_eia_na(fgd_df).pipe(pudl.helpers.convert_to_date)
+
+    # Convert thousands of dollars to dollars
+    fgd_df.loc[:, fgd_df.columns.str.endswith("_1000_dollars")] *= 1000
+    fgd_df.columns = fgd_df.columns.str.replace("_1000_dollars", "")  # Rename columns
+
+    # Convert SO2 test date column to datetime
+    # This column only exists for 2008-2011
+    # First, convert a few troublesome datetimes that look like m/yy or mm/yy
+    if not fgd_df.so2_test_date.isnull().all():  # If column not empty
+        troublesome_dates = fgd_df.so2_test_date.str.contains(
+            r"^[0-9]{1,2}\/[0-9]{2}$", regex=True, na=False
+        )
+        logger.info(
+            f"Rescuing troublesome dates: {fgd_df[troublesome_dates].so2_test_date.unique()}"
+        )
+        fgd_df.loc[troublesome_dates, "so2_test_date"] = fgd_df[
+            troublesome_dates
+        ].so2_test_date.str.pad(5, fillchar="0")
+        fgd_df.loc[troublesome_dates, "so2_test_date"] = pd.to_datetime(
+            fgd_df.loc[troublesome_dates, "so2_test_date"], format="%m/%y"
+        )
+
+    test_datetime = pd.to_datetime(
+        fgd_df.so2_test_date, format="mixed", errors="coerce", dayfirst=False
+    )
+    dropped_dates = fgd_df[
+        (test_datetime.isnull()) & (fgd_df.so2_test_date.notnull())
+    ].so2_test_date.unique()
+    logger.info(f"Dropping SO2 test dates that were not parseable: {dropped_dates}")
+    fgd_df.loc[:, "so2_test_date"] = test_datetime
+
+    # Handle mixed boolean types in control flag column
+    fgd_df = convert_col_to_bool(
+        df=fgd_df,
+        col_name="fgd_control_flag",
+        true_values=["Y", 1.0],
+        false_values=["N", 0.0],
+    )
+
+    # Take the non-NA values from each column for duplicate rows
+    pkey = ["plant_id_eia", "so2_control_id_eia", "report_date"]
+    fgd_df = pudl.helpers.dedupe_and_drop_nas(fgd_df, primary_key_cols=pkey)
```

Comment:
TODO (for me): I'll use this method for the EIA923 cooling systems information stuff after this gets merged.

Reply:
I can just do it in this PR, no trouble